src/main/java/edu/hm/hafner/analysis/parser/xmlparser/XmlParser.java

```diff
@@ -0,0 +1,206 @@
+     * @param mapper = Tag to Property Mapper.
+     */
+    public XmlParser(final String root, final Map<String, String> mapper) {
+        this.xmlIssueRoot = root;
+        this.propertyMapper = mapper;
+    }
+
+    /**
+     * Create a new ReportParser object.
+     */
+    public XmlParser() {
+        this(DEFAULT_ROOT_PATH, new DefaultXmlMapper());
+    }
+
+
+    @Override
+    public Report parse(final ReaderFactory readerFactory) {
+        try {
+            Document doc = readerFactory.readDocument();
+
+            XPathFactory xPathFactory = XPathFactory.newInstance();
+            XPath path = xPathFactory.newXPath();
+
+            IssueBuilder issueBuilder = new IssueBuilder();
+            Report report = new Report();
+
+            NodeList issues = (NodeList)path.evaluate(getXmlIssueRoot(), doc, XPathConstants.NODESET);
+            // Read all issues.
+            for (Element issue : XmlElementUtil.nodeListToList(issues)) {
+                issueBuilder
+                        .setFileName(notNullableRead(path, FILE_NAME, issue))
+                        .setLineStart(notNullableRead(path, LINE_START, issue))
+                        .setLineEnd(notNullableRead(path, LINE_END, issue))
+                        .setColumnStart(notNullableRead(path, COLUMN_START, issue))
+                        .setColumnEnd(notNullableRead(path, COLUMN_END, issue))
+                        .setCategory(notNullableRead(path, CATEGORY, issue))
+                        .setType(notNullableRead(path, TYPE, issue))
+                        .guessSeverity(notNullableRead(path, SEVERITY, issue))
+                        .setMessage(notNullableRead(path, MESSAGE, issue))
+                        .setDescription(notNullableRead(path, DESCRIPTION, issue))
+                        .setPackageName(notNullableRead(path, PACKAGE_NAME, issue))
+                        .setModuleName(notNullableRead(path, MODULE_NAME, issue))
+                        .setOrigin(notNullableRead(path, ORIGIN, issue))
+                        .setReference(notNullableRead(path, REFERENCE, issue))
+                        .setFingerprint(notNullableRead(path, FINGERPRINT, issue))
+                        .setAdditionalProperties(notNullableRead(path, ADDITIONAL_PROPERTIES, issue));
+                // TODO LineRange is missing.
+                report.add(issueBuilder.build());
+
+            }
+            return report;
+        }
+        catch (ParsingException | XPathExpressionException e) {
+            throw new ParsingException(e);
+        }
+    }
+
+    /**
+     * Read a value from XML and set a default value if its null.
+     * @param path = xpath.
+     * @param value = value which should be read.
+     * @param issue = current issue.
+     * @return value or default value.
+     * @throws XPathExpressionException error.
+     */
+    private String notNullableRead(final XPath path, final String value, final Element issue) throws XPathExpressionException {
```

Comment:
I think it is save to return `null` here as the `IssueBuilder` instance correctly handles `null` values.